libraries/Microsoft.Bot.Builder.Azure.Blobs/BlobsTranscriptStore.cs

```diff
@@ -0,0 +1,419 @@
+                {
+                    var blobClient = _containerClient.Value.GetBlobClient(bl.Name);
+                    return await GetActivityFromBlobClientAsync(blobClient).ConfigureAwait(false);
+                })
+                .Select(t => t.Result)
+                .ToArray();
+
+            if (pagedResult.Items.Length == PageSize)
+            {
+                pagedResult.ContinuationToken = blobs.Last().Name;
+            }
+
+            return pagedResult;
+        }
+
+        /// <summary>
+        /// List conversations in the channelId.
+        /// </summary>
+        /// <param name="channelId">Channel Id.</param>
+        /// <param name="continuationToken">Continuatuation token to page through results.</param>
+        /// <returns>A <see cref="Task"/> A task that represents the work queued to execute.</returns>
+        public async Task<PagedResult<TranscriptInfo>> ListTranscriptsAsync(string channelId, string continuationToken = null)
+        {
+            const int PageSize = 20;
+
+            if (string.IsNullOrEmpty(channelId))
+            {
+                throw new ArgumentNullException($"missing {nameof(channelId)}");
+            }
+
+            string token;
+            List<TranscriptInfo> conversations = new List<TranscriptInfo>();
+            do
+            {
+                token = null;
+
+                var resultSegment = _containerClient.Value.GetBlobsAsync(BlobTraits.Metadata, prefix: $"{SanitizeKey(channelId)}/")
+                                    .AsPages(token);
+
+                await foreach (var blobPage in resultSegment)
+                {
+                    foreach (BlobItem blobItem in blobPage.Values)
+                    {
+                        // Unescape the Id we escaped when we saved it
+                        var conversation = new TranscriptInfo() { Id = Uri.UnescapeDataString(blobItem.Name.Split('/').Where(s => s.Length > 0).Last()), ChannelId = channelId };
+                        if (continuationToken != null)
+                        {
+                            if (conversation.Id == continuationToken)
+                            {
+                                // we found continuation token
+                                continuationToken = null;
+                            }
+
+                            // skip record
+                        }
+                        else
+                        {
+                            conversations.Add(conversation);
+                            if (conversations.Count == PageSize)
+                            {
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+            while (!string.IsNullOrEmpty(token) && conversations.Count < PageSize);
+
+            var pagedResult = new PagedResult<TranscriptInfo>();
+            pagedResult.Items = conversations.ToArray();
```

Comment:
Can use the initializer: new PageResult<TranscriptInfo>{ Items = conversations.ToArray() };

Reply:
fixed